### ISSUE TITLE
Branchprotector is failing to update repos in the kubeflow org.

### ISSUE THREAD

**Issue by cjwagner** [kubernetes/test-infra#18397]
**What happened**:
Branchprotector started failing between 
[1278462001802121218](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-test-infra-branchprotector/1278462001802121218) | 2020-07-01 22:54:51 +0000 UTC 
[1278446903859613702](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-test-infra-branchprotector/1278446903859613702) | 2020-07-01 21:54:52 +0000 UTC

with an error message that suggests many (all?) updates to the kubeflow repo have failed due to 404s while GETing the existing branch protection settings from GH.

<details><summary>Full error message</summary>{"component":"branchprotector","error":"update kubeflow: [update tf-operator: [update v0.7-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update v0.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.3-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.4-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.2-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.5-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update kubeflow: [update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update terrytangyuan-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.2-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.6-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update pdmack-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update pipelines from protected=false: get current branch protection: getting branch protection 404: Not Found, update pr/4909 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.4-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.5-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.8-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update v0.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.3-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update reporting: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update testing: [update agwl-logging from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found], update examples: [update amygdala-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update cookbook from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update remove_wip from protected=false: get current branch protection: getting branch protection 404: Not Found, update xgboost_ames from protected=false: get current branch protection: getting branch protection 404: Not Found, update xgboost_zillow from protected=false: get current branch protection: getting branch protection 404: Not Found], update example-seldon: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update community: [update labels-owners from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found], update pytorch-operator: [update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update v0.3-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.4-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.5-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update caffe2-operator: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update katib: [update 0.6 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update 0.2 from protected=false: get current branch protection: getting branch protection 404: Not Found], update website: [update zhenghuiwang-patch-2 from protected=false: get current branch protection: getting branch protection 404: Not Found, update sarahmaddox-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.6-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update will-may from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-patch-3 from protected=false: get current branch protection: getting branch protection 404: Not Found, update ewilderj-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update theadactyl-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update use-cases-doc from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.5-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-metawatcher from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-pgke-cert-mgr from protected=false: get current branch protection: getting branch protection 404: Not Found, update sarahmaddox-fix-link from protected=false: get current branch protection: getting branch protection 404: Not Found, update onprem-installation from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.2-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.3-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.4-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found], update kubebench: [update v0.5-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update Akado2009-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update Akado2009-reviewer from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.3-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.4-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update batch-predict: [update ankush-signing-out from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found], update pipelines: [update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update release-1.0 from protected=true: get current branch protection: getting branch protection 404: Not Found, update Marketplace---Only-use-CloudSQL-when-provided from protected=false: get current branch protection: getting branch protection 404: Not Found, update avolkov/Kubeflow-TFX-pipeline-with-lightweight-python-deployer-component from protected=false: get current branch protection: getting branch protection 404: Not Found, update avolkov/Samples/Notebooks---Added-Lightweight-component-sharing-notebooks from protected=false: get current branch protection: getting branch protection 404: Not Found, update avolkov/Stopped-printing-huge-amount-of-infor-to-the-notebook from protected=false: get current branch protection: getting branch protection 404: Not Found], update internal-acls: [update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update ankush-signing-out from protected=false: get current branch protection: getting branch protection 404: Not Found], update mpi-operator: update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update crd-validation: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update marketing-materials: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update chainer-operator: update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update arena: [update release-v0.1.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update release-v0.3.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update release-v0.4.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update feature/supportPrune from protected=false: get current branch protection: getting branch protection 404: Not Found, update features/pytorch-backup from protected=false: get current branch protection: getting branch protection 404: Not Found, update features/pytorch from protected=false: get current branch protection: getting branch protection 404: Not Found, update mpi_operator from protected=false: get current branch protection: getting branch protection 404: Not Found, update release-v0.2.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update release-v0.5.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update release-v0.6.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update tfjobv1-0.6.0.rc0 from protected=false: get current branch protection: getting branch protection 404: Not Found], update mxnet-operator: [update release-notes from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found], update fairing: update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update features: update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update homebrew-core: [update formulae-brew-sh-actions from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found], update manifests: [update terrytangyuan-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-test-v1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update pdmack-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update richardsliu-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update update_centraldashboard_ from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.6-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.8-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update zhenghuiwang-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7-branch from protected=true: get current branch protection: getting branch protection 404: Not Found, update automated-cherry-pick-of-#949-upstream-v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update metadata: [update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update v0.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update kfserving: [update v0.4.0 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1beta1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update 0.2.2 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.3.0 from protected=false: get current branch protection: getting branch protection 404: Not Found], update common: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update xgboost-operator: [update terrytangyuan-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found], update code-intelligence: [update auto-update from protected=false: get current branch protection: getting branch protection 404: Not Found, update hamelsmu-remove-embedding-site from protected=false: get current branch protection: getting branch protection 404: Not Found, update ingress from protected=false: get current branch protection: getting branch protection 404: Not Found, update test-triage-action from protected=false: get current branch protection: getting branch protection 404: Not Found, update update-0703 from protected=false: get current branch protection: getting branch protection 404: Not Found, update auto-update-manual from protected=false: get current branch protection: getting branch protection 404: Not Found, update edit-readme from protected=false: get current branch protection: getting branch protection 404: Not Found, update hamelsmu/add-batching-utility from protected=false: get current branch protection: getting branch protection 404: Not Found, update move-notebook from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update correct-links from protected=false: get current branch protection: getting branch protection 404: Not Found], update kfctl: [update v1.0-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update richardsliu-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update v0.7-branch from protected=false: get current branch protection: getting branch protection 404: Not Found], update kfp-tekton: [update kfp-backend-merge from protected=false: get current branch protection: getting branch protection 404: Not Found, update status-update from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update install-use from protected=false: get current branch protection: getting branch protection 404: Not Found], update frontend: update master from protected=true: get current branch protection: getting branch protection 404: Not Found, update .github: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update gcp-blueprints: [update v1.1-branch from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=true: get current branch protection: getting branch protection 404: Not Found], update community-infra: [update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update jun from protected=false: get current branch protection: getting branch protection 404: Not Found], update kfp-tekton-backend: [update dev-instructions from protected=false: get current branch protection: getting branch protection 404: Not Found, update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update project-intent from protected=false: get current branch protection: getting branch protection 404: Not Found], update fate-operator: update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update blog: [update master from protected=false: get current branch protection: getting branch protection 404: Not Found, update terrytangyuan-patch-1 from protected=false: get current branch protection: getting branch protection 404: Not Found, update add-author-links from protected=false: get current branch protection: getting branch protection 404: Not Found, update add-comments from protected=false: get current branch protection: getting branch protection 404: Not Found, update docs from protected=false: get current branch protection: getting branch protection 404: Not Found, update fastpages-automated-upgrade from protected=false: get current branch protection: getting branch protection 404: Not Found, update gh-pages from protected=false: get current branch protection: getting branch protection 404: Not Found]]","file":"prow/cmd/branchprotector/protect.go:143","func":"main.main","level":"error","msg":"0","severity":"error","time":"2020-07-20T19:27:52Z"}</details>

**Please provide links to example occurrences, if any**:
https://k8s-testgrid.appspot.com/sig-testing-misc#branchprotector&width=20

**Anything else we need to know?**:

I suspect this is caused by insufficient permissions granted to `k8s-ci-robot` for the kubeflow org. `k8s-ci-robot` must be an admin in order to change branch protection settings for all repos in the org. This probably results in a 404 rather than a 403 to deter unauthorized probing that could confirm the existence of a repo or branch.

The failures started after configuring the branchprotection config for kubeflow: https://github.com/kubernetes/test-infra/pull/18095

/assign @Bobgy @jlewi 
/cc @michelle192837

**Comment by Bobgy** [comment]
@jlewi Can you add required permissions to @k8s-ci-robot for kubeflow org? I don't think I have permissions to do so.

**Comment by Bobgy** [comment]
@cjwagner is there a need to rollback #18095 quickly?

**Comment by cjwagner** [comment]
@Bobgy The job will still succeed for other orgs and repos so this doesn't block functionality, but it does cause noise for oncall since the job is failing and this could also obscure other issues.  If the permissions can't be fixed quickly please roll back your changes and we can reapply them when the appropriate permissions have been configured.

**Comment by jlewi** [comment]
> it does cause noise for oncall since the job is failing and this could also obscure other issues

@cjwagner LGTM to rollback until we can fix this on our end and figure out a long term solution.

@Bobgy repo permissions are granted via GitOps using periobolous so you could configure an appropriate PR.
https://github.com/kubeflow/internal-acls/blob/f6e2bdb4a92b1ae6b808f2c7b81ab1077acfa69a/github-orgs/kubeflow/org.yaml#L672

Does admin permission at the org level need to be granted? Granting org level admin privileges to the bot account or event repo admin privileges is a significant escalation of privileges. I'm ok doing it if it makes sense but I'd like to make sure we understand the implications and do so intentionally. Can you open up an appropriate issue somewhere in Kubeflow to track that.

**Comment by cjwagner** [comment]
Yes org level admin permissions are required for things like organization webhook management and team management. It also provides the needed repo level permissions without having to grant the bot access to each repo individually.
